keras_cv/models/mobilenet_v3.py

```
@@ -0,0 +1,534 @@
+
+    def apply(x):
+        shortcut = x
+        prefix = "expanded_conv/"
+        infilters = backend.int_shape(x)[channel_axis]
+
+        if block_id:
+            prefix = "expanded_conv_{}/".format(block_id)
+
+            x = layers.Conv2D(
+                Depth()(infilters * expansion),
+                kernel_size=1,
+                padding="same",
+                use_bias=False,
+                name=prefix + "expand",
+            )(x)
+            x = layers.BatchNormalization(
+                axis=channel_axis,
+                epsilon=1e-3,
+                momentum=0.999,
+                name=prefix + "expand/BatchNorm",
+            )(x)
+            x = activation(x)
+
+        x = layers.DepthwiseConv2D(
+            kernel_size,
+            strides=stride,
+            padding="same" if stride == 1 else "valid",
+            use_bias=False,
+            name=prefix + "depthwise",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=channel_axis,
+            epsilon=1e-3,
+            momentum=0.999,
+            name=prefix + "depthwise/BatchNorm",
+        )(x)
+        x = activation(x)
+
+        if se_ratio:
+            with custom_object_scope({"hard_sigmoid": HardSigmoid()}):
+                x = SqueezeAndExcite2D(
+                    filters=Depth()(infilters * expansion),
+                    ratio=se_ratio,
+                    squeeze_activation="relu",
+                    excite_activation="hard_sigmoid",
+                )(x)
+
+        x = layers.Conv2D(
+            filters,
+            kernel_size=1,
+            padding="same",
+            use_bias=False,
+            name=prefix + "project",
+        )(x)
+        x = layers.BatchNormalization(
+            axis=channel_axis,
+            epsilon=1e-3,
+            momentum=0.999,
+            name=prefix + "project/BatchNorm",
+        )(x)
+
+        if stride == 1 and infilters == filters:
```

is there ever a case where infilters != filters that is a valid case?

---

if not, lets raise an error in that case

---

Yes, infilters != filters would still be a valid case